### PR TITLE
[Minor][EDA-1386] Integrate fixes from demo

### PR DESCRIPTION
**Pull request by ecrespillo-eb**
- Added bug fixes finded before the demo.
- Completely removed Game class.
- Added a function for challenge creation.
- Refactor of ServerEvent class and inheritance in server_event.py module.
- Tests update.
- Constants update.

### CODE REVIEW

File: server/utilities_server_event.py [hunk 1]
```diff
@@ -13,63 +14,82 @@
...
-        asyncio.create_task(penalize(data, game_name))
+class ServerEvent:
+    def __init__(self, response, client):
+        self.response = response
+        self.client = client
 
-    async def search_value(self, response, client, value):
-        value_search = response.get('data', {}).get(value)
+    async def search_value(self, value):
+        value_search = self.response.get('data', {}).get(value)
```
Comment (axelfernandez-eb) [inline server/utilities_server_event.py]: Maybe here we can use a Constant?

File: server/utilities_server_event.py [hunk 2]
```diff
@@ -13,63 +14,82 @@
...
-    async def game_over(self, game: dict, data):
-        await notify_end_game_to_client(
-            game.get('players'),
-            data.turn_data,
-        )
-        end_data = end_data_for_web(data.turn_data)
+    async def game_over(self, data, game: dict):
+        next_turn(data.game_id)
+        end_data = make_end_data_for_web(data.turn_data)
+        await notify_end_game_to_client(game.get('players'), data.turn_data)
```
Comment (axelfernandez-eb) [inline server/utilities_server_event.py]: same here with players string?

File: tests/test_router.py [hunk 3]
```diff
@@ -5,32 +5,19 @@
...
-            TIME_CHALLENGE,
-        )
-        mock_notify_challenge.assert_awaited_once_with('Pepe', 'Ana', challenge_id)
+        with patch('server.router.make_challenge') as mock_make_challenge:
+            async with AsyncClient(app=app, base_url="http://test") as ac:
+                response = await ac.post(
+                    "/challenge",
+                    json=data
+                )
+        mock_make_challenge.assert_awaited_once_with(['Ana', 'Pepe'])
```
Comment (axelfernandez-eb) [inline tests/test_router.py:20]: Nice Refactor!! 🥇

File: server/utilities_server_event.py (hunk 2 above)
Comment (ecrespillo-eb) [reply server/utilities_server_event.py]: Done!

File: server/utilities_server_event.py (hunk 1 above)
Comment (ecrespillo-eb) [reply server/utilities_server_event.py]: Done!

**Review by axelfernandez-eb — APPROVED** [review]

**Pull request merged**